File: server/Timelogger/Entities/Project.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Timelogger.Entities
{
    public class Project
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int ProjectStatusId { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime EndDate { get; set; }

    }
}

File: client/src/app/types/project.ts
import "./status";
import { Status } from "./status";
export interface Project {
  id: number;
  name: string;
  description: string;
  projectStatusId: number;
  startDate: Date;
  endDate: Date;
}

export interface ProjectStatus {
  id: number;
  name: string;
  description: string;
  status: Status;
  statusName: string;
  startDate: Date;
  endDate: Date;
}
